core/src/main/java/org/infinispan/reactive/publisher/impl/LocalPublisherManagerImpl.java

```diff
@@ -404,7 +406,10 @@ private int getNextSegment(PrimitiveIterator.OfInt segmentIter) {
       AdvancedCache<K, V> cache = deliveryGuarantee == DeliveryGuarantee.AT_MOST_ONCE ? this.cache : remoteCache;
       return handleSpecificObjects(parallelPublisher, keysToInclude, keysToExclude, keyFlowable ->
             // Filter out all the keys that aren't in the cache
-            keyFlowable.filter(cache::containsKey)
+            keyFlowable.concatMapMaybe(key ->
+               RxJavaInterop.completionStageToMaybe(cache.getAsync(key)
+                     .thenApply(value -> value == null ? null : key))
```

Comment:
Can't we use `containsKeyAsync`?

Reply:
Yes, I didn't realize we had that :)

Reply:
Although, sadly the code will be not much prettier since we still have to use `concatMapSingle`.